### PR TITLE
skip ipv6 if not supported

### PR DESCRIPTION
**Pull request by chipitsine**
travis ci does not support ipv6, the best we can do, skip it

### CODE REVIEW

File: t/058-tcp-socket.t [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # vim:set ft= ts=4 sw=4 et fdm=marker:
 
+
```
Comment (agentzh) [inline t/058-tcp-socket.t]: Why this change?

File: t/058-tcp-socket.t [hunk 2]
```diff
@@ -3298,6 +3299,7 @@ failed to receive a line: closed []
 close: 1 nil
 --- no_error_log
 [error]
+--- skip_eval: 3: system("ping6 -c 1 ::1 >/dev/null 2>&1") ne 0
```
Comment (agentzh) [inline t/058-tcp-socket.t]: Maybe we should make this nicer via `--- use_ipv6` in `Test::Nginx::Socket`?

File: t/058-tcp-socket.t (hunk 1 above)
Comment (chipitsine) [inline t/058-tcp-socket.t]: I've forgotten to have a look at "git diff", do not pay attention to that line

File: t/058-tcp-socket.t (hunk 2 above)
Comment (chipitsine) [inline t/058-tcp-socket.t]: do you mean skip all tests in 058 ? only single test fails.

Comment (chipitsine) [inline t/058-tcp-socket.t]: anyway, I'm not very familiar with Test:Nginx (I'm just studying that framework), you can use whatever you think is better

Comment (agentzh) [inline t/058-tcp-socket.t]: @chipitsine Other nginx modules of openresty have test cases against IPv6 as well. So better abstract it away? We can check if `--with-ipv6` is present in the `nginx -V` output instead of calling `ping6`.

Comment (agentzh) [inline t/058-tcp-socket.t]: @chipitsine `--- use_ipv6` is just my idea. It does not exist yet. Needs patches for `Test::Nginx` :)

Comment (chipitsine) [inline t/058-tcp-socket.t]: sounds good

Comment (chipitsine) [inline t/058-tcp-socket.t]: can we merge my dirty hack for now?
we can polish it later if we will have enough time and will

**Comment by agentzh** [comment]
Merged with modifications.
